docs/conventions/wizards.md

```diff
@@ -0,0 +1,218 @@
+[All Conventions](/docs/conventions.md) / Wizards
+
+# Wizards
+
+> Use wizards to build multi-step forms that help users complete complex tasks
+
+## File Structure
+
+This directory tree represents a wizard and its steps:
+
+```
+app/
+└── wizards/
+    └── placements/
+        ├── add_placement_wizard.rb
+        └── add_placement_wizard/
+            ├── check_your_answers_step.rb
+            ├── mentors_step.rb
+            └── subject_step.rb
+```
+
+## How to build a wizard
+
+This is a simplified example to demonstrate the various parts that come together to make a wizard.
+
+### 1. Create a wizard class
+
+```ruby
+# app/wizards/placements/add_placement_wizard.rb
+module Placements
+  class AddPlacementWizard < BaseWizard
+    def define_steps
+      # Define the wizard steps here
+      add_step(SubjectStep)
+      add_step(MentorsStep)
+      add_step(CheckYourAnswersStep)
+    end
+  end
+end
+```
+
+### 2. Create step classes
+
+Think of step objects as a hybrid between form objects and decorators:
+
+- **Form objects** – to hold attributes and validate their values
+- **Decorators** – just like a ViewComponent's HTML markup comes coupled with a Ruby class, or how helpers and decorators are used by the view layer
+
+```ruby
+# app/wizards/placements/add_placement_wizard/subject_step.rb
+class Placements::AddPlacementWizard::SubjectStep < Placements::BaseStep
+  attribute :subject_id
+  validates :subject_id, presence: true
+
+  def subject_name
+    Subject.find(subject_id).name
+  end
+end
+
+# app/wizards/placements/add_placement_wizard/mentors_step.rb
+class Placements::AddPlacementWizard::MentorsStep < Placements::BaseStep
+  attribute :mentor_ids
+  validates :mentor_ids, presence: true
+end
+
+# app/wizards/placements/add_placement_wizard/check_your_answers_step.rb
+class Placements::AddPlacementWizard::CheckYourAnswersStep < Placements::BaseStep
+  # This step has no attributes of its own
+end
+```
+
+### 3. Create view partials for each step
+
+Each step comes coupled with a view partial.
+
+```erb
+# views/placements/wizards/add_placement_wizard/_subject_step.html.erb
+<%= form_for(subject_step, url: current_step_path, method: :put) do |f| %>
+  <%= f.govuk_error_summary %>
+
+  <%= f.govuk_collection_radio_buttons(
+    :subject_id,
+    Subject.all,
```

Comment:
Would we use `subject_step.subjects_for_selection` as best practice here as it would be tested within the step?